cmd/end2endtest/ballot.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"go.vocdoni.io/dvote/apiclient"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+func init() {
+	ops["ballotelection"] = operation{
+		test:        &E2EBallotElection{},
+		description: "ballot election with unique values, maxCount, maxValue, maxTotalCost to test different ballotProtocol configurations",
+		example:     os.Args[0] + " --operation=ballotelection --votes=1000",
+	}
+}
+
+var _ VochainTest = (*E2EBallotElection)(nil)
+
+type E2EBallotElection struct {
+	e2eElection
+}
+
+func (t *E2EBallotElection) Setup(api *apiclient.HTTPclient, c *config) error {
+	t.api = api
+	t.config = c
+
+	p := &models.Process{
+		StartBlock: 0,
+		BlockCount: 100,
+		Status:     models.ProcessStatus_READY,
+		EnvelopeType: &models.EnvelopeType{
+			EncryptedVotes: false,
+			UniqueValues:   true},
+		CensusOrigin: models.CensusOrigin_OFF_CHAIN_TREE_WEIGHTED,
+		VoteOptions: &models.ProcessVoteOptions{
+			MaxCount:     2,
+			MaxValue:     6,
+			MaxTotalCost: 10,
+			CostExponent: 1,
+		},
+		Mode: &models.ProcessMode{
+			AutoStart:     true,
+			Interruptible: true,
+		},
+		MaxCensusSize: uint64(t.config.nvotes),
+	}
+
+	if err := t.setupElection(nil, p); err != nil {
+		return err
+	}
+
+	log.Debugf("election details: %+v", *t.election)
+	return nil
+}
+
+func (t *E2EBallotElection) Teardown() error {
+	// nothing to do here
+	return nil
+}
+
+func (t *E2EBallotElection) Run() error {
+	c := t.config
+	api := t.api
+
+	voteValues, expectedR := t.randomBallotVotes()
+
+	// convert expected results int slice to string slice to be used on compare the results later
+	resultsChan := make(chan [][]string)
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		resultsChan <- sliceIntToStr(expectedR)
+	}()
+
+	results := <-resultsChan
+	wg.Wait()
```

Comment:
also, this whole block looks like something you did while debugging the expectedResults bug. (i remember there was a time.Sleep here as well)
i must say it's a very peculiar way of doing `results := sliceIntToStr(expectedR)`
and as i said, you really should make `expectedResults` a slice of BigInts, so you don't even need that sliceIntToStr conversion,
and you can delete this whole thing entirely and just use `expectedR` (renamed to `expectedResults` for consistency)